kernel/src/tbfheader.rs

```diff
@@ -351,6 +401,36 @@ impl TbfHeader {
             _ => (0, 0),
         }
     }
+
+    /// Get the address in RAM this process was specifically compiled for. If
+    /// the process is position independent, return `None`.
+    crate fn get_fixed_address_ram(&self) -> Option<u32> {
+        match *self {
+            TbfHeader::TbfHeaderV2(hd) => hd.fixed_addresses.map_or(None, |fa| {
+                if fa.start_process_ram == 0xFFFFFFFF {
+                    None
+                } else {
+                    Some(fa.start_process_ram)
+                }
+            }),
+            _ => None,
+        }
```

Comment:
This implementation feels a bit awkward to me. How about the following?

```
        let hd = match self {
            TbfHeader::TbfHeaderV2(hd) => hd,
            _ => return None,
        };
        match hd.fixed_addresses.as_ref()?.start_process_ram {
            0xFFFFFFFF => None,
            start => Some(start),
        }
```

Reply:
I will take your word for it!